torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -551,6 +557,7 @@ ProcessGroupNCCL::ProcessGroupNCCL(
   blockingWait_ = parseEnvVarFlag(NCCL_BLOCKING_WAIT);
   asyncErrorHandling_ = parseEnvVarFlag(NCCL_ASYNC_ERROR_HANDLING);
   desyncDebug_ = parseEnvVarFlag(NCCL_DESYNC_DEBUG);
+  avoidRecordStreams_ = parseEnvVarFlag(NCCL_AVOID_RECORD_STREAMS);
```

Comment:
Nit: can we prefix env's with `TORCH_DISTRIBUTED_` instead of `NCCL_`? People sometimes think these are NCCL env's, so we are starting to avoid that.
Also, do we plan to maintain this env for the longer term?